test/runtests.jl

```diff
@@ -22,6 +22,11 @@ using Test
     @inferred a1(1.0)
 end
 
+@testset "fix(::Type, ...)" begin
+    f = @inferred fix(CartesianIndex, nothing, Some(1))
```

Comment:
This `@inferred` passes even without this change. Is it worth clarifying what it's trying to test?

Reply:
I think its good to have this test. While this does not go wrong in master it is something that one can imagine going wrong. So this test makes future refactoring easier.

Reply:
I was wondering what could go wrong that would make this test fail. I agree the test should stay.